File: .agents/trustgraph-viewer/lib/trustgraphConfig.ts
import fs from "fs";
import path from "path";

type TrustGraphConfig = {
  neo4jUri: string;
  neo4jUser: string;
  neo4jPassword: string;
  chromaHost: string;
  chromaPort: number;
};

function readLocalEnv(): Record<string, string> {
  const envPath = path.resolve(process.cwd(), "../trustgraph.env");
  if (!fs.existsSync(envPath)) return {};

  return fs
    .readFileSync(envPath, "utf-8")
    .split(/\r?\n/)
    .reduce<Record<string, string>>((acc, rawLine) => {
      const line = rawLine.trim();
      if (!line || line.startsWith("#") || !line.includes("=")) return acc;

      const [rawKey, ...rawValue] = line.split("=");
      const key = rawKey.trim();
      const value = rawValue.join("=").trim().replace(/^['"]|['"]$/g, "");
      if (key) acc[key] = value;
      return acc;
    }, {});
}

export function getTrustGraphConfig(): TrustGraphConfig {
  const localEnv = readLocalEnv();

  return {
    neo4jUri: process.env.NEO4J_URI || localEnv.NEO4J_URI || "bolt://localhost:7687",
    neo4jUser: process.env.NEO4J_USER || localEnv.NEO4J_USER || "neo4j",
    neo4jPassword: process.env.NEO4J_PASSWORD || localEnv.NEO4J_PASSWORD || "trustgraph_secret",
    chromaHost: process.env.CHROMA_HOST || localEnv.CHROMA_HOST || "localhost",
    chromaPort: Number(process.env.CHROMA_PORT || localEnv.CHROMA_PORT || "8800"),
  };
}

File: .agents/adapters/trustgraph_config.py
#!/usr/bin/env python3
"""Shared TrustGraph runtime configuration for local adapters."""

import base64
import os
from pathlib import Path


def _load_local_env() -> None:
    env_path = Path(__file__).resolve().parents[1] / "trustgraph.env"
    if not env_path.exists():
        return

    for raw_line in env_path.read_text(encoding="utf-8").splitlines():
        line = raw_line.strip()
        if not line or line.startswith("#") or "=" not in line:
            continue

        key, value = line.split("=", 1)
        os.environ.setdefault(key.strip(), value.strip().strip('"').strip("'"))


_load_local_env()

NEO4J_HTTP_ENDPOINT = os.getenv(
    "TRUSTGRAPH_NEO4J_HTTP_ENDPOINT",
    os.getenv("NEO4J_HTTP_ENDPOINT", "http://localhost:7474/db/neo4j/tx/commit"),
)
NEO4J_BOLT_URI = os.getenv("NEO4J_URI", "bolt://localhost:7687")
NEO4J_USER = os.getenv("NEO4J_USER", "neo4j")
NEO4J_PASSWORD = os.getenv("NEO4J_PASSWORD", "trustgraph_secret")

CHROMA_HOST = os.getenv("CHROMA_HOST", "localhost")
CHROMA_PORT = int(os.getenv("CHROMA_PORT", "8800"))
CHROMA_COLLECTION = os.getenv("CHROMA_COLLECTION", "codebase")


def neo4j_auth_header() -> str:
    token = f"{NEO4J_USER}:{NEO4J_PASSWORD}".encode("utf-8")
    encoded = base64.b64encode(token).decode("ascii")
    return f"Basic {encoded}"
